cookbooks/tomography_based_plate_motions/plugins/tomography_based_plate_motions.cc

```diff
@@ -1350,6 +1528,23 @@ namespace aspect
                              "We can evaluate the effects of lithospheric thickness variations on the surface plate "
                              "motions using this. Currently, this parameter sets the lithosphere to a constant "
                              "thickness of 100 km.");
+          prm.declare_entry ("Use slab2 database", "false",
+                             Patterns::Bool (),
+                             "This parameter value determines if we want to the slab depths and thickness defined "
+                             "in Slab2 database (Hayes et al., 2018). Currently, it only supports input as a 2D   "
+                             "grid on the boundary with columns representing depths and thickness of the slabs.");
+          prm.declare_entry ("Slab temperature anomaly", "-650",
+                             Patterns::Double(),
+                             "The temperature anomaly of the slabs relative to the surrounding mantle. Since we use the diffused "
+                             "temperature anomaly in our current implementation, the default value is approximately the difference "
+                             "between the surface and mantle temperature."
+                             "Units:  \\si{\\kelvin}");
+          prm.declare_entry ("Trench weak zone thickness", "50e3",
+                             Patterns::Double(0),
+                             "This parameter value determines the thickness of weak zone representing the oceanic trenches. "
+                             "In our implementation, we set these weak zones parallel to the top of the slabs in the slab2 model "
+                             "and therefore, this parameter only makes sense when we are using the slab2 models. "
```

Comment:
```suggestion
                             "and therefore, this parameter is only used when we are using the slab2 model. "
```

In addition, can you be consistent in how slab2 is spelled (Slab2? Slab2.0?)